packages/typescript/src/lib/Typescript.ts

```diff
@@ -190,17 +198,17 @@ export class FederatedTypesPlugin {
 
     try {
       // Try to resolve exposed component using index
-      const files = fs.readdirSync(path.join(rootDir, entry));
+      const files = this.getFiles(path.join(rootDir, entry));
```

Comment:
I think it would be best if we check `path.join(rootDir, entry)` is actually a directory or not, before traversing it. Previously this wouldn't have been a problem since we are not caching it.

Since, we are caching this now, we might end up with having a entry in `fsCache` map which in reality is never a directory.


I'm also thinking that we might actually never end up with a scenario where we have to traverse with `index` file. I wanted to this scenario but couldn't do it. If you don't mind, can you do a quick test if this `try` block will actually resolve without throwing an error.

Reply:
This is not a situation where we traverse with index. This is when the `index.tsx?` file is inferred. For example, I don't expose `Button/index.tsx`, rather I expose `Button`. That's what is covered in the try scenario.

So the button path is added to the cache which is still needed so we can read the directory contents and grab the proper extension for the index file